internal/models/quizzes/quizzes.go

```diff
@@ -135,52 +135,51 @@ func GetQuizzes(db *sql.DB) ([]Quiz, error) {
 	}
 	defer rows.Close()
 
-	quizzes := []Quiz{}
-	for rows.Next() {
-		var quiz Quiz
-		var imageURL sql.NullString
-		err := rows.Scan(
-			&quiz.ID,
-			&quiz.Title,
-			&imageURL,
-			&quiz.AvailableFrom,
-			&quiz.AvailableTo,
-			&quiz.CreatedAt,
-			&quiz.LastModifiedAt,
-			&quiz.Published,
-			&quiz.IsDeleted,
-		)
-		if err != nil {
-			return nil, err
-		}
-
-		// Set image URL
-		tempURL, err := data_handling.ConvertNullStringToURL(&imageURL)
-		if err != nil {
-			return nil, err
-		}
-		quiz.ImageURL = *tempURL
-
-		quizzes = append(quizzes, quiz)
-	}
-	return quizzes, nil
+	return scanQuizzesFromFullRows(rows)
 }
 
+// Get all the quizzes that are not published and not deleted.
 func GetNonPublishedQuizzes(db *sql.DB) ([]Quiz, error) {
```

Comment:
instead of 2 functions, getnonpublisehd and getallpublished, maybe do 1 function that takes a bool parameter (published?)

Reply:
[bff05b3](https://github.com/Molnes/Nyhetsjeger/pull/86/commits/bff05b32337ecee00f12e1786b29c6f085b6d648)